facia-tool/app/tools/FaciaToolArchive.scala

```diff
@@ -0,0 +1,54 @@
+package tools
+
+import com.amazonaws.regions.{Regions, Region}
+import com.amazonaws.services.dynamodbv2.AmazonDynamoDBAsyncClient
+import com.amazonaws.services.dynamodbv2.model._
+import common.{ExecutionContexts, Logging}
+import conf.{Switches, Configuration}
+import org.joda.time.{DateTimeZone, DateTime}
+import play.api.libs.json._
+import scala.collection.JavaConverters._
+import awswrappers.dynamodb._
+
+import scala.util.{Failure, Success}
+
+case class ArchiveRequest(email: String, updateJson: JsValue, diff: JsValue)
+
+object FaciaToolArchive extends ExecutionContexts with Logging {
+  val TableName = "FaciaToolUpdateHistory"
+  private val dynamoClient: Option[AmazonDynamoDBAsyncClient] =
+    if (Configuration.environment.isProd) {
+      val c = new AmazonDynamoDBAsyncClient()
+      c.setRegion(Region.getRegion(Regions.EU_WEST_1))
+      Option(c)
+    } else None
+
+  def dayKey(date: DateTime) = date.toString("yyyy-MM-dd")
```

Comment:
Writes. We are doing a lot on the CSS report though, probably far more than you'll see through Facia Tool, so it might be OK in this case.

The latency of something like Kinesis doesn't make it a good candidate to feed directly into your main DB I don't think. You want a DB that's optimised for storing your data in the way you want to access it (for which Dynamo is a good candidate), but also to push updates as events to Kinesis, for any replication, backups, analytics, etc.